src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+  const uniqueValues: any[] = [];
```

Comment:
The `uniqueValues` array is being used to track unique keys, but it's type is defined as `any[]`, which is not type-safe. Consider using a set of type `Set<keyof T>` to automatically handle unique values without needing to check with `includes`.